hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/SCMContainerPlacementCapacity.java

```diff
@@ -98,7 +98,7 @@ public SCMContainerPlacementCapacity(final NodeManager nodeManager,
    * @throws SCMException  SCMException
    */
   @Override
-  public List<DatanodeDetails> chooseDatanodes(
+  protected List<DatanodeDetails> chooseDatanodesInternal(
```

Comment:
At line 105, I think you need to change `super.chooseDatanodes` to `super.chooseDatanodesInternal()`.

Also to get the test to pass, I needed to add this to TestSCMContainerPlacementCapacity:

```
    when(mockNodeManager.getNodeByUuid(anyString())).thenAnswer(
        invocation -> {
          String uuid = invocation.getArgument(0);
          return datanodes.stream().filter(
              datanode -> datanode.getUuid().toString().equals(uuid)).findFirst()
                  .orElse(null);
        });
```